Project.toml

```diff
@@ -54,6 +55,7 @@ ArgParse = "1"
 ArtifactWrappers = "0.2"
 AtmosphericProfilesLibrary = "0.1"
 CLIMAParameters = "0.7"
+CUDA = "4"
```

Comment:
If this PR adds/updates the `compat` entry, should we also bump another ClimaAtmos release? (I'm thinking for the downstream update, e.g., the Coupler)

Reply:
This PR's scope is related to enabling GPU jobs. We can bump the patch release in a separate PR.